elftools/construct/macros.py

```diff
@@ -632,3 +632,24 @@ def OnDemandPointer(offsetfunc, subcon, force_build = True):
 
 def Magic(data):
     return ConstAdapter(Field(None, len(data)), data)
+
+class StreamOffset(Construct):
```

Comment:
`construct` is vendored, so we should not have changes in the `elftools/construct` dir. If we need this adapter, it should be defined outside (in pyelftools's own code)

Reply:
I'll probably reuse this for ranges, so common/utils?